presentation/src/main/java/bg/government/virusafe/app/utils/validators/EgnValidator.kt

```diff
@@ -0,0 +1,122 @@
+@file:Suppress("MagicNumber")
+package bg.government.virusafe.app.utils.validators
+
+import bg.government.virusafe.app.utils.empty
+import com.upnetix.applicationservice.registration.model.Gender
+import org.threeten.bp.LocalDate
+import org.threeten.bp.temporal.ChronoUnit
+import java.util.*
+
+/**
+ * Validate EGN.
+ */
+private const val EGN_MOD = 11
+
+class EgnValidator : PersonalIdValidator {
+
+	private val egnWeights = listOf(2, 4, 8, 5, 10, 9, 7, 3, 6)
+	private var personalNumber: String? = null
+	private var isValid = false
+
+	override var years: Long? = null
+		private set
+
+	override var gender: String? = null
+		private set
+
+	override fun isValidPersonalId(): Boolean = isValid
+
+	override fun initPersonalNumber(personalNumber: String?) {
+		this.personalNumber = personalNumber
+		isValid = isValidPersonalNumber()
+		if (!isValid) {
+			years = null
+			gender = null
+		}
+	}
+
+	override fun clearAll() {
+		personalNumber = null
+		isValid = false
+		years = null
+		gender = null
+	}
+
+	@Suppress("ReturnCount")
+	private fun isValidPersonalNumber(): Boolean {
+		if (personalNumber == null || personalNumber == String.empty) {
+			return true
+		} else if (personalNumber!!.length != 10) {
```

Comment:
Avoid using the not-null assertion operator (!!) use safe call operator instead (?.)